config.go

```diff
@@ -2,127 +2,372 @@ package wtsc
+		errors = append(errors, "chain_pool")
+	}
+
+	if !IsValidServicerList(cfg.ServicerKeys) {
+		errors = append(errors, "servicer_keys")
+	}
+
+	if cfg.StakeWeight < 1 || cfg.StakeWeight > 4 {
+		errors = append(errors, "stake_weight")
+	}
+
+	if cfg.MinIncreasePercent <= 1 && cfg.MinIncreasePercent > 100 {
+		errors = append(errors, "min_increase_percent")
+	}
+
+	if !IsValidMinServiceStake(cfg.MinServiceStake) {
+		errors = append(errors, "min_service_stake")
+	}
+
+	if cfg.TimePeriod < 6 || cfg.TimePeriod > 48 {
+		errors = append(errors, "time_period")
+	}
+
+	if !IsEmptyString(cfg.ResultsPath) && !IsWritableDirectory(cfg.ResultsPath) {
+		// empty string disable the feature so it's ok been empty
+		errors = append(errors, "results_path")
+	}
+
+	if _, err := zerolog.ParseLevel(cfg.LogLevel); err != nil {
+		errors = append(errors, "log_level")
+	}
+
+	if cfg.LogFormat != LogTextFormat && cfg.LogFormat != LogJsonFormat {
+		errors = append(errors, "log_format")
+	}
+
+	if _, err := cron.ParseStandard(cfg.Schedule); err != nil {
+		errors = append(errors, "schedule")
+	}
+
+	if cfg.MaxWorkers <= 0 {
+		errors = append(errors, "max_workers")
+	}
+
+	if !IsValidHttpURI(cfg.PocketRPC) {
+		errors = append(errors, "pocket_rpc")
+	}
+
+	if cfg.MaxRetries < 0 {
+		errors = append(errors, "max_retries")
+	}
+
+	// less than 1s is not allowed
+	if cfg.MaxTimeout < 1000 {
+		errors = append(errors, "max_retries")
+	}
+
+	valid = true
+
+	return
 }
 
 // LoadConfig loads and returns the configuration by reading the `configPath` file.
 func LoadConfig() *Config {
 	cfg := Config{}
-	var configPath = "./config.json"
+	var configPath = GetConfigFilePath()
+	Logger.Info().Str("path", configPath).Msg("reading config file")
 	var jsonFile *os.File
 	var bz []byte
 	if _, err := os.Stat(configPath); err != nil && os.IsNotExist(err) {
-		log.Fatal().Str("configPath", configPath).Msg("config file not found")
+		Logger.Fatal().Str("configPath", configPath).Msg("config file not found")
 	}
 	// reopen the file to read into the variable
 	jsonFile, err := os.OpenFile(configPath, os.O_RDONLY|os.O_CREATE, os.ModePerm)
 	if err != nil {
-		log.Fatal().Str("configPath", configPath).Err(err).Msg("failed to open config file")
+		Logger.Fatal().Str("configPath", configPath).Err(err).Msg("failed to open config file")
 	}
 	bz, err = io.ReadAll(jsonFile)
 	if err != nil {
-		log.Fatal().Str("configPath", configPath).Err(err).Msg("failed to read config file")
+		Logger.Fatal().Str("configPath", configPath).Err(err).Msg("failed to read config file")
 	}
 	err = jsonFile.Close()
 	if err != nil {
-		log.Fatal().Str("configPath", configPath).Err(err).Msg("failed to close config file")
+		Logger.Fatal().Str("configPath", configPath).Err(err).Msg("failed to close config file")
 	}
 	err = json.Unmarshal(bz, &cfg)
 	if err != nil {
-		log.Fatal().Str("configPath", configPath).Err(err).Msg("failed to unmarshal config file")
-	}
-
-	err = ValidateConfig(&cfg)
```

Comment:
was moved out on every place it is called, but now I move it at the very end of LoadConfig, previously was not trigger Fatal() on hot reload of config file to prevent process stop due to a bad change.